coil-base/src/main/java/coil/bitmap/RealBitmapPool.kt

```diff
@@ -92,24 +92,25 @@ internal class RealBitmapPool(
         return getDirtyOrNull(width, height, config) ?: createBitmap(width, height, config)
     }
 
-    @Synchronized
     override fun getDirtyOrNull(@Px width: Int, @Px height: Int, config: Bitmap.Config): Bitmap? {
         require(!config.isHardware) { "Cannot create a mutable hardware bitmap." }
 
-        val result = strategy.get(width, height, config)
-        if (result == null) {
-            logger?.log(TAG, Log.VERBOSE) { "Missing bitmap=${strategy.stringify(width, height, config)}" }
-            misses++
-        } else {
-            bitmaps -= result
-            currentSize -= result.allocationByteCountCompat
-            hits++
-            normalize(result)
-        }
+        synchronized(this) {
```

Comment:
Whats the advantage to using this `synchronized` vs. the annotation on the method?

Reply:
It's pretty minor, but using `synchronized` lets us grab the lock after we've checked if the config is `HARDWARE`. `@Synchronized` is just syntactic sugar that wraps the method in a `synchronized` block

Reply:
Actually, ended up reverting this part since the benefit is probably imperceptibly small.